core/src/sigverify_shreds.rs

```diff
@@ -152,8 +144,7 @@ fn get_slot_leaders(
             };
             let leader = leaders.entry(slot).or_insert_with(|| {
                 let leader = leader_schedule_cache.slot_leader_at(slot, Some(bank))?;
-                // Discard the shred if the slot leader is the node itself.
-                (&leader != self_pubkey).then_some(leader)
+                Some(leader)
```

Comment:
I think this and the previous line can just be simplified to 

`leader_schedule_cache.slot_leader_at(slot, Some(bank))`